xmtp_mls/src/groups/sync.rs

```diff
@@ -424,6 +460,7 @@ where
             let group_id = self.group_id.as_slice();
             let message_id =
                 get_message_id(encoded_payload_bytes.as_slice(), group_id, timestamp_ns);
```

Comment:
This is still calling the deprecated `get_message_id`. Isn't that going to cause duplicate messages? If not, maybe we should include an explanatory comment

Reply:
AFAIK the alternative method for generating message ids via `calculate_message_id()` is used for application messages, and not here for  transript messages.   I'll be the first to say it's a bit jank to have two different methods for generating ids.  But these transcript messages won't have idempotency keys to use as part of the message id generation.  Unless we wanted to say that all messages of all types get an idempotency key.  Thoughts on that?

Reply:
That’s fair, it’s a bit of a weird case. I’d be ok with giving transcript messages an idempotency key (maybe the commit message timestamp) just for consistency

Reply:
We are now using the same function, `calculate_message_id()` throughout.  I thought that would introduce breaking changes unrelated to this work, but apparently not.